docs/docs/features/studio.md

```diff
@@ -0,0 +1,109 @@
+---
+title: ZMK Studio
+sidebar_label: Studio
+---
+
+:::warning[Alpha Feature]
+
+ZMK Studio support is in alpha at this point. Although our best efforts are being made to keep compatibility during active development, but is not guaranteed.
+
+:::
+
+ZMK Studio provides runtime update functionality to ZMK powered devices. Studio is still under active development, and is not yet ready for casual end user use.
+
+## Building
+
+Building for ZMK Studio involves two main additional items.
+
+- Build with the `studio-rpc-usb-uart` snippet to enable the endpoint used for Studio communication over USB.
+- Enable the `ZMK_STUDIO` Kconfig setting.
+
+### GitHub Actions
+
+First add a `studio-rpc-usb-uart` to the `snippet` property of your build configuration. For a split keyboard, you should do this _only_ for your central/left side, e.g.:
+
+```
+---
+include:
+  - board: nice_nano_v2
+    shield: corne_left
+    snippet: studio-rpc-usb-uart
+  - board: nice_nano_v2
+    shield: corne_right
+```
+
+Next, enable the `ZMK_STUDIO` Kconfig symbol, for example by adding the following line to your .conf file:
+
+```
+CONFIG_ZMK_STUDIO=y
+```
+
+### Local Build
+
+When building locally, use the `-S` parameter to include the `studio-rpc-usb-uart` snippet, and append the `ZMK_STUDIO` Kconfig as an additional CMake argument, e.g.:
+
+```bash
+west build -d build/cl_studio -b nice_nano_v2 \
+  -S studio-rpc-usb-uart -- -DSHIELD=corne_left -DCONFIG_ZMK_STUDIO=y
+```
+
+## Behaviors
+
+Normally, ZMK will only build and include behaviors that are referenced by your keymap, and unused behavior will be skipped. However, studio builds using the `studio-rpc-usb-uart` snippet will automatically define the `ZMK_BEHAVIORS_KEEP_ALL` value, which changes the approach and builds all possible behaviors into the firmware.
+
+A few controls are available to override this behavior for fine-grained control of what behaviors are built. Behaviors can be kept or omitted by defining certain values in the top of your keymap file before the standard `behaviors.dtsi` file is included.
```

Comment:
#I think we need to provide a reference for what defines can be used. If we don't want to pollute here, we can just drop a link to a header file?

Reply:
They aren't defined by default. They only exists if the user defines them, which triggers the omit/keep functionality.

Reply:
What I mean is that the user needs some way to figure out what `name` in `ZMK_BEHAVIORS_OMIT_##_name` corresponds to for a given behavior.

Reply:
Oh, fully agreed! We just sadly can't point to a header for this. Still feels like this should be either on each behavior page, or maybe in the behaviors index?

Reply:
Right, there doesn't seem to be a single file where that can be checked. Do you think we could simply use the behavior reference to avoid adding a new naming convention? e.g. 
`ZMK_BEHAVIORS_OMIT_mt` for `&mt`. (I think partial lower case might be acceptable given the tradeoffs.)

Reply:
Let me ponder that one... I have mixed feelings on that on. Just a convention of "capitalize the normal reference" seems easy enough.